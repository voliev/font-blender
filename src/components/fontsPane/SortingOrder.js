import React from "react";
import ArrowDownwardRounded from "@material-ui/icons/ArrowDownwardRounded";
import ArrowUpwardRounded from "@material-ui/icons/ArrowUpwardRounded";
import PropTypes from "prop-types";
import "./styles/SortingOrder.css";

const SortingOrder = ({ sortingOrderChangeHandler, value }) => (
  <div className="SortingOrder">
    <button
      className="btn btn-round-big"
      onClick={sortingOrderChangeHandler}
      value={value}
      title="Change sorting order"
    >
      {value === "asc" ? <ArrowDownwardRounded /> : <ArrowUpwardRounded />}
    </button>
  </div>
);

SortingOrder.propTypes = {
  sortingOrderChangeHandler: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired
};

export default SortingOrder;
