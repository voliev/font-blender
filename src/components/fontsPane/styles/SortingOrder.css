.SortingOrder {
  align-items: center;
  display: flex;
  height: 4rem;
  width: 4rem;
}

.SortingOrder .btn {
  background-color: transparent;
  color: var(--blue);
}
