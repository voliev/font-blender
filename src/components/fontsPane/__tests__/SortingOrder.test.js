import React from "react";
import { mount, shallow } from "enzyme";
import ArrowDownwardRounded from "@material-ui/icons/ArrowDownwardRounded";
import ArrowUpwardRounded from "@material-ui/icons/ArrowUpwardRounded";
import SortingOrder from "../SortingOrder";

const setup = (renderFn = shallow, propsOverrides = {}) => {
  const props = {
    value: "asc",
    sortingOrderChangeHandler: jest.fn(),
    ...propsOverrides
  };

  return {
    ...props,
    wrapper: renderFn(<SortingOrder {...props} />)
  };
};

afterEach(() => {
  jest.clearAllMocks();
});

it("should render without crashing", () => {
  const { wrapper } = setup();

  expect(wrapper).toMatchSnapshot();
});

it("should receive corresponding props", () => {
  const { wrapper, value, sortingOrderChangeHandler } = setup(mount);

  expect(typeof wrapper.prop("value")).toBe("string");
  expect(wrapper.prop("value")).toBe(value);
  expect(typeof wrapper.prop("sortingOrderChangeHandler")).toBe("function");
  expect(wrapper.prop("sortingOrderChangeHandler")).toBe(
    sortingOrderChangeHandler
  );
});

it("should set corresponding value on initial render", () => {
  const { wrapper } = setup(mount);

  expect(wrapper.prop("value")).toBe("asc");
});

it("should call handler parameter on button click", () => {
  const { wrapper, sortingOrderChangeHandler } = setup();

  const event = {
    currentTarget: { value: "des" }
  };
  wrapper.find("button").simulate("click", event);

  expect(sortingOrderChangeHandler).toHaveBeenCalled();
  expect(sortingOrderChangeHandler).toHaveBeenCalledTimes(1);
  expect(sortingOrderChangeHandler).toHaveBeenCalledWith(event);
});

it("should set corresponding icon depending on value", () => {
  const { wrapper } = setup();

  expect(wrapper.find(ArrowDownwardRounded).length).toBe(1);
  expect(wrapper.find(ArrowUpwardRounded).length).toBe(0);

  wrapper.setProps({ value: "des" });

  expect(wrapper.find(ArrowDownwardRounded).length).toBe(0);
  expect(wrapper.find(ArrowUpwardRounded).length).toBe(1);
});
