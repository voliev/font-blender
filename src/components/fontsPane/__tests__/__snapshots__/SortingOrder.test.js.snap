// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
<div
  className="SortingOrder"
>
  <button
    className="btn btn-round-big"
    onClick={[MockFunction]}
    title="Change sorting order"
    value="asc"
  >
    <ArrowDownwardRoundedIcon />
  </button>
</div>
`;
